src/main/java/gyro/aws/ec2/TransitGatewayRouteTableResource.java

```diff
@@ -0,0 +1,270 @@
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+import gyro.aws.AwsResource;
+import gyro.aws.Copyable;
+import gyro.core.GyroUI;
+import gyro.core.Type;
+import gyro.core.Wait;
+import gyro.core.resource.Id;
+import gyro.core.resource.Output;
+import gyro.core.scope.State;
+import gyro.core.validation.Required;
+import software.amazon.awssdk.services.ec2.Ec2Client;
+import software.amazon.awssdk.services.ec2.model.CreateTransitGatewayRouteTableResponse;
+import software.amazon.awssdk.services.ec2.model.DescribeTransitGatewayRouteTablesResponse;
+import software.amazon.awssdk.services.ec2.model.Ec2Exception;
+import software.amazon.awssdk.services.ec2.model.Filter;
+import software.amazon.awssdk.services.ec2.model.TransitGatewayRoute;
+import software.amazon.awssdk.services.ec2.model.TransitGatewayRouteTable;
+import software.amazon.awssdk.services.ec2.model.TransitGatewayRouteTableAssociation;
+import software.amazon.awssdk.services.ec2.model.TransitGatewayRouteTablePropagation;
+import software.amazon.awssdk.services.ec2.model.TransitGatewayRouteTableState;
+
+/**
+ * Creates a transit gateway route table.
+ *
+ * Example
+ * -------
+ *
+ * .. code-block:: gyro
+ *
+ *    aws::transit-gateway-route-table transit-gateway-route-table-example
+ *        transit-gateway: $(aws::transit-gateway example-transit-gateway)
+ *
+ *        association
+ *            peering-attachment: $(aws::transit-gateway-peering-attachment transit-gateway-peering-attachment-example)
+ *        end
+ *
+ *        route
+ *            destination-cidr-block: "10.0.0.0/24"
+ *            peering-attachment: $(aws::transit-gateway-peering-attachment transit-gateway-peering-attachment-example)
+ *        end
+ *    end
+ */
+@Type("transit-gateway-route-table")
+public class TransitGatewayRouteTableResource extends Ec2TaggableResource<TransitGatewayRouteTable>
+    implements Copyable<TransitGatewayRouteTable> {
+
+    private TransitGatewayResource transitGateway;
+    private List<TransitGatewayRouteTableAssociationResource> association;
+    private List<TransitGatewayRouteTablePropagationResource> propagation;
+    private List<TransitGatewayRouteResource> route;
+
+    // Output
+
+    private String id;
+
+    /**
+     * The transit gateway of the route table. (Required)
+     */
+    @Required
+    public TransitGatewayResource getTransitGateway() {
+        return transitGateway;
+    }
+
+    public void setTransitGateway(TransitGatewayResource transitGateway) {
+        this.transitGateway = transitGateway;
+    }
+
+    /**
+     * The attachments to associate with the route table.
+     *
+     * @subresource gyro.aws.ec2.TransitGatewayRouteTableAssociationResource
+     */
+    public List<TransitGatewayRouteTableAssociationResource> getAssociation() {
+        if (association == null) {
+            association = new ArrayList<TransitGatewayRouteTableAssociationResource>();
```

Comment:
Redundant to initialize an array list with type

```suggestion
            association = new ArrayList<>();
```